TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -93,6 +99,17 @@ public class BadgeCommands : ICommandCollection
             };
         }
 
+        /// <summary>
+        /// Convineintly handles many optional arguments for badge commands. May throw an exception when given faulty form informaton, so this should be surrounded by a try/catch
+        /// </summary>
+        private (Badge.BadgeSource?, string?, bool?) InterpretBadgeInfoArgs(Optional<Badge.BadgeSource> sourceOpt, Optional<Form> formOpt, Optional<Shiny> shinyOpt)
+        {
+            Badge.BadgeSource? source = sourceOpt.IsPresent ? sourceOpt.Value : null;
+            string? form = formOpt.IsPresent ? formOpt.Value.Name : null;
+            bool? shiny = shinyOpt.IsPresent ? shinyOpt.Value : false;
+            return (source, form, shiny);
```

Comment:
This looks equivalent to
```
return (sourceOpt.OrElse(null), formOpt.OrElse(null), shinyOpt.OrElse(false));
```

Reply:
also is defaulting to shiny=false correct? It's still nullable, which indicates to me you wanted it to be kind of a tri-state of "yes", "no" and "don't care whether shiny or not"